pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1750,3 +1770,86 @@ func (r *ReconcileClusterDeployment) deleteStaleProvisions(provs []*hivev1.Clust
 		}
 	}
 }
+
+// getAllSyncSetInstances return all syncset instances for a cluster deployment
+func (r *ReconcileClusterDeployment) getAllSyncSetInstances(cd *hivev1.ClusterDeployment) ([]*hivev1.SyncSetInstance, error) {
+
+	list := &hivev1.SyncSetInstanceList{}
+	err := r.Client.List(context.TODO(), list, client.InNamespace(cd.Namespace))
+	if err != nil {
+		return nil, err
+	}
+
+	syncSetInstances := []*hivev1.SyncSetInstance{}
+	for i, syncSetInstance := range list.Items {
+		fmt.Println("syncsetinstance name: ", syncSetInstance.Name)
+		if syncSetInstance.Spec.ClusterDeployment.Name == cd.Name {
+			syncSetInstances = append(syncSetInstances, &list.Items[i])
+		}
+	}
+
+	return syncSetInstances, nil
+}
+
+// checkFailureSyncCondition return true when syncsetinstance status condition has hivev1.ApplyFailureSyncCondition
+func checkFailureSyncCondition(syncSetInstances []*hivev1.SyncSetInstance) bool {
+	for _, syncSetInstance := range syncSetInstances {
+		for _, condition := range syncSetInstance.Status.Conditions {
+			fmt.Println("Condition Type: ", condition.Type)
+			if condition.Type == hivev1.ApplyFailureSyncCondition {
```

Comment:
You need to check `condition.Status` as well. When the syncsetinstance is applied successfully, the syncsetinstance will still have this condition, but it will have a status of false.